enhancements/builds/node-cache-for-builds.md

```diff
@@ -0,0 +1,212 @@
+additional image store into its own working container storage within the running build container.
+This ensures that any layers pruned from underneath the container during the build will not break
+the running build. Since builds in 4.4 and earlier pull every layer into the build pod, the impact
+of copying these layers should be with respect to ephemeral storage.
+
+#### Builds can modify content in the node's container storage
+
+*Risk:* Writing to the node's container storage is a security risk, can lead to the kubelet running
+rogue images.
+
+*Mitigation:* Node container storage will be mounted as read-only as additional read-only storage.
+
+#### Unbound image cache growth
+
+*Risk:* Cached images can grow to the point that nodes fail due to disk pressure.
+
+*Mitigation:* Node cache is read-only and managed by the kubelet and CRI-O. Builds take advantage of
+the overlay filesystem and write their layers to ephemeral storage within the image build container.
+When the build completes, the container's ephemeral storage is recycled.
+
+#### Bypass permission checks to pull an image
+
+*Risk:* Builds can bypass permission checks that prevent a user from pulling an image.
+
+*Mitigation:* Builds can specify `forcePull` via the `BuildConfig` spec. As a part of this
+enhancement, we will restore the `forcePull` option for cluster-wide `BuildDefaults` and
+`BuildOverrides`. This will allow cluster admins to require every build to pull the manifests for an
+image from its upstream repository.
+
+#### Containers/storage upgrades can make builds unusable
+
+*Risk:* On upgrade, version skew between the `containers/storage` library used by cri-o and buildah
+can cause builds to be unusable.
+
+*Mitigation:* The image layer format in use by `containers/storage` is designed to be backwards
+compatible between major versions. If in the event data in the additional store is not usable,
+buildah should fall back to its default behavior of pulling image content from the upstream
+registry.
+
+#### Host path with image cache does not exist
+
+*Risk:* The node where the build runs does not have a `/var/lib/containers/storage` cache.
+
+*Mitigation:* In the unlikely event that the cri-o storage is not configured on the node (or is
+placed in an alternate location), the `HostPath` mount will use the `DirectoryOrCreate` type to
+ensure that such a directory always exists on the host. Buildah will fall back to its current
+behavior if the additional storage directory is empty.
+
+#### Build container does not have permission to read the image cache
+
+*Risk:* Build container does not have permission to read the image cache on the node.
+
+*Mitigation:* Builds already run as a privileged container, and thus are able to obtain more
+privileges than the parent kubelet. If for any reason the node does not allow privileged containers
+to read contents in the node image cache, buildah should be able to fall back to its current
+behavior.
```

Comment:
@nalind another item that we should verify - if the additional store can't be read by buildah, will it default to "no additional store?"